LEAF_Request_Portal/sources/Email.php

```diff
@@ -777,10 +771,91 @@ function attachApproversAndEmail(int $recordID, int $emailTemplateID, mixed $log
+        } elseif ($emailTemplateID === -7) {
+            $recordInfo = $this->getRecord($recordID);
+            $comments = $this->getDeletedComments($recordID);
+
+            $comment = $comments[0]['comment'] === '' ? '' : 'Reason for cancelling: ' . $comments[0]['comment'] . '<br /><br />';
+
+            $title = strlen($recordInfo[0]['title']) > 45 ? substr($recordInfo[0]['title'], 0, 42) . '...' : $recordInfo[0]['title'];
+
+            $this->addSmartyVariables(array(
+                "truncatedTitle" => $title,
+                "fullTitle" => $recordInfo[0]['title'],
+                "recordID" => $recordID,
+                "service" => $recordInfo[0]['service'],
+                "lastStatus" => $comments[0]['actionType'],
+                "siteRoot" => $this->siteRoot,
+                "field" => null,
+                "comment" => $comment
+            ));
+
+            $this->processPriorStepsEmailed($this->getPriorStepsEmailed($recordID));
+            $this->setTemplateByID($emailTemplateID);
+            $this->sendMail($recordID);
         } elseif ($emailTemplateID > 1) {
             $return_value = $this->sendMail($recordID); // Check for custom event to finalize email on Notify Next
         }
 
         return $return_value;
     }
+
+    private function getRecord(int $recordID): array
+    {
+        $vars = array(':recordID' => $recordID);
+        $strSQL =  "SELECT `rec`.`userID`, `rec`.`serviceID`, `ser`.`service`, `rec`.`title`,
+                        `rec`.`lastStatus`
+                    FROM `records` AS `rec`
+                    LEFT JOIN `services` AS `ser` USING (`serviceID`)
+                    WHERE `recordID` = :recordID";
+
+        $recordInfo = $this->portal_db->prepared_query($strSQL, $vars);
+
+        return $recordInfo;
+    }
+
+    private function getDeletedComments(int $recordID): array
+    {
+        $vars = array(':recordID' => $recordID);
+        $strSQL =  "SELECT `comment`, `actionType`
+                    FROM `action_history`
+                    WHERE `recordID` = :recordID
+                    AND `actionType` = 'deleted'
+                    ORDER BY `actionID` DESC";
+
+        $recordInfo = $this->portal_db->prepared_query($strSQL, $vars);
+
+        return $recordInfo;
+    }
+
+    private function getPriorStepsEmailed(int $recordID): array
+    {
+        // get the email_tracker data for this record
+        $vars = array(':recordID' => $recordID);
+        $sql = 'SELECT `recipients`
+                FROM `email_tracker`
+                WHERE `recordID` = :recordID';
+
+        $return_result = $this->portal_db->prepared_query($sql, $vars);
+
+        return $return_result;
+    }
+
+    private function processPriorStepsEmailed(array $email_recipients): void
+    {
+        $recipient_list = array();
+
+        foreach ($email_recipients as $recipient) {
+            $clean_list = str_replace('Recipient(s): ', '', $recipient['recipients']);
```

Comment:
Since we're introducing a dependency on the recipient string, there should be a comment on line ~369 where it's assigned to explain that the string has a specific format and shouldn't be changed.

Ideally the list of recipients should be maintained independent of any other text. Addressing this can be scoped to another story.

Reply:
Comment added. I also created a Jira ticket #4388 to address the ideal situation.